### PR TITLE
Jakarta Annotations 2.1

### PR DESCRIPTION
**Pull request by m0mus**
Signed-off-by: Dmitry Kornilov <dmitry.kornilov@oracle.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   https://projects.eclipse.org/projects/ee4j.ca/releases/2.1
   - [x] Generated IP Log
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/annotation/jakarta.annotation-api/2.1.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-annotations-tck-2.1.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/common-annotations-api/issues/98
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by ivargrimstad** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by m0mus** [comment]
I am opening this PR for review. One note, there is no compatible implementation for Jakarta Annotations. So, I am leaving the compatible implementation check box unchecked.

**Comment by ivargrimstad** [comment]
> I am opening this PR for review. One note, there is no compatible implementation for Jakarta Annotations. So, I am leaving the compatible implementation check box unchecked.

So, in order for the ballot to start we will need to find someone that provides an implementation, e.g. GlassFish?

**Comment by ivargrimstad** [comment]
Looks good! 
Only thing that is missing is a CI and a corresponding CCR

**Comment by m0mus** [comment]
> Looks good! Only thing that is missing is a CI and a corresponding CCR

It's done.

File: annotations/2.1/_index.md [hunk 1]
```diff
@@ -15,6 +15,17 @@ The JDK version required will be aligned with Jakarta EE 10.
...
 
 * [Jakarta Annotations 2.1 Release Record](https://projects.eclipse.org/projects/ee4j.ca/releases/2.1)
+    * [Jakarta EE Platform 10 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee10/JakartaEE10ReleasePlan)
+* [Jakarta Annotations 2.1 Specification Document](./annotations-spec-2.1.pdf) (PDF)
+* [Jakarta Annotations 2.1 Specification Document](./annotations-spec-2.1.html) (HTML)
+* [Jakarta Annotations 2.1 Javadoc](./apidocs)
+* [Jakarta Annotations 2.1 TCK](https://download.eclipse.org/jakartaee/annotations/2.1/jakarta-annotations-tck-2.1.0.zip)  ([sig](https://download.eclipse.org/jakartaee/annotations/2.1/jakarta-annotations-tck-2.1.0.zip.sig),  [sha](https://download.eclipse.org/jakartaee/annotations/2.1/jakarta-annotations-tck-2.1.0.zip.sha256),  [pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
+* Maven coordinates
+    * [jakarta.annotation:jakarta.annotation-api:jar:2.1.0](https://search.maven.org/artifact/jakarta.annotation/jakarta.annotation-api/2.1.0/jar)
+
```
Comment (ivargrimstad) [inline annotations/2.1/_index.md:25]: Add:
```
* Compatible Implementations used for [ratification](https://www.eclipse.org/projects/efsp/?version=1.2#efsp-ratification).
  * [Eclipse GlassFish 7.0.1-M1](https://github.com/eclipse-ee4j/glassfish/releases/download/7.0.0-M1/glassfish-7.0.0-M1.zip)
```

**Comment by ivargrimstad** [comment]
The SHA-256 doesn't match https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-annotations-tck-2.1.0-tckinfo.txt

Should be: '875811dbc18353f74ff96afb7ced8fb6b4cfa02daac07806e6701f698e3a9d4b`

Applies to the CCR and the TCK Results page

**Review by ivargrimstad — CHANGES_REQUESTED** [review]
- Add the Compatible implementation under "Compatible Implementation used for ratification" as well as under Compatible Implementations.
- Update the SHA-256 for the TCK results and in the CCR
- Otherwise, everything looks fine!

**Comment by m0mus** [comment]
@ivargrimstad All issues are addressed. Please re-review.

**Review by ivargrimstad — APPROVED** [review]
Looks good now, thanks @m0mus!
I'll get the ballot started shortly

**Comment by ivargrimstad** [comment]
Please delete the file .idea/misc.xml

**Comment by m0mus** [comment]
> Please delete the file .idea/misc.xml

Oooops... 
Deleted

**Pull request merged**